homeassistant/components/swiss_public_transport/__init__.py

```diff
@@ -78,9 +80,7 @@ async def async_migrate_entry(
 
     if config_entry.minor_version == 1:
         # Remove wrongly registered devices and entries
-        new_unique_id = (
-            f"{config_entry.data[CONF_START]} {config_entry.data[CONF_DESTINATION]}"
-        )
+        new_unique_id = unique_id_from_config(config_entry.data)
```

Comment:
This addition of `via` should probably bump the config entry version again as it's not really backwards compatible.

Reply:
Do I understand correctly that it is forward compatible and does not need a migration but it going back will break. Is that correct? @gjohansson-ST

Reply:
I wonder if this doesn't need to bump the major version as with the addition of `via` in the unique id I don't think it would work if a user goes backwards in versions.
I don't think bumping a minor version is enough.

Reply:
I am not that familiar with this kind of policy within home assistant. Currently I only add via stations, but there are some more useful options in the api which will require another jump in the future. What would you suggest? Be on the save side and jump a major? @gjohansson-ST

Reply:
@gjohansson-ST We already had a discussion about that here: https://github.com/home-assistant/core/pull/107087#discussion_r1443688094

My interpretation would be to make sure one cannot bump back to a prior version as it would break and therefore bump a **major** version.

Reply:
I also extended the unit test for the migration to be able to test all different starting versions using parametrisation.